plugins/obs-filters/color-correction-filter.c

```diff
@@ -0,0 +1,364 @@
+			filter->diag.y, filter->b_line.x);
+	vec3_mulf(&filter->rotational_matrix_row2,
+			&filter->rotational_matrix_row2, 2.0f);
+
+	vec3_set(&filter->rotational_matrix_row3, filter->b_line.y,
+			filter->a_line.x, filter->diag.z);
+	vec3_mulf(&filter->rotational_matrix_row3,
+			&filter->rotational_matrix_row3, 2.0f);
+}
+
+static void color_correction_filter_destroy(void *data)
+{
+	struct color_correction_filter_data *filter = data;
+
+	if (filter->effect) {
+		obs_enter_graphics();
+		gs_effect_destroy(filter->effect);
+		obs_leave_graphics();
+	}
+
+	bfree(data);
+}
+
+static void *color_correction_filter_create(obs_data_t *settings,
+	obs_source_t *context)
+{
+	struct color_correction_filter_data *filter =
+		bzalloc(sizeof(struct color_correction_filter_data));
+	char *effect_path = obs_module_file("color_correction_filter.effect");
+
+	filter->context = context;
+
+	/* My way of making these as close to a const as possible. */
+	filter->root3 = 0.57735f;
+	vec3_set(&filter->half_unit, 0.5f, 0.5f, 0.5f);
+
+	/* Set and clear all vectors. */
+	vec3_zero(&filter->rotational_matrix_row1);
+	vec3_zero(&filter->rotational_matrix_row2);
+	vec3_zero(&filter->rotational_matrix_row3);
+	vec3_zero(&filter->rotational_quaternion);
+	vec3_zero(&filter->cross);
+	vec3_zero(&filter->square);
+	vec3_zero(&filter->wimag);
+	vec3_zero(&filter->diag);
+	vec3_zero(&filter->a_line);
+	vec3_zero(&filter->b_line);
+
+	obs_enter_graphics();
+
+	filter->effect = gs_effect_create_from_file(effect_path, NULL);
+	if (filter->effect) {
+		filter->gamma_param = gs_effect_get_param_by_name(
+				filter->effect, SETTING_GAMMA);
+		filter->color_curve_param = gs_effect_get_param_by_name(
+				filter->effect, SETTING_COLORCURVE);
+		filter->contrast_param = gs_effect_get_param_by_name(
+				filter->effect, SETTING_CONTRAST);
+		filter->brightness_param = gs_effect_get_param_by_name(
+				filter->effect, SETTING_BRIGHTNESS);
+		filter->saturation_param = gs_effect_get_param_by_name(
+				filter->effect, SETTING_SATURATION);
+		filter->opacity_param = gs_effect_get_param_by_name(
+				filter->effect, SETTING_OPACITY);
+
+		/* Pre-Computes. */
+		filter->rotational_matrix_row1_param =
+				gs_effect_get_param_by_name(
+				filter->effect, SETTING_ROTMATRIXROW1);
+		filter->rotational_matrix_row2_param =
+				gs_effect_get_param_by_name(
+				filter->effect, SETTING_ROTMATRIXROW2);
+		filter->rotational_matrix_row3_param =
+				gs_effect_get_param_by_name(
+				filter->effect, SETTING_ROTMATRIXROW3);
+	}
+
+	obs_leave_graphics();
+
+	bfree(effect_path);
+
+	if (!filter->effect) {
+		color_correction_filter_destroy(filter);
+		return NULL;
+	}
+
+	color_correction_filter_update(filter, settings);
+	return filter;
+}
+
+static void color_correction_filter_render(void *data, gs_effect_t *effect)
+{
+	struct color_correction_filter_data *filter = data;
+
+	if (!obs_source_process_filter_begin(filter->context, GS_RGBA,
+		OBS_ALLOW_DIRECT_RENDERING))
```

Comment:
If you check plugins/obs-filters/color-filter.c for its [equivalent line](https://github.com/jp9000/obs-studio/blob/master/plugins/obs-filters/color-filter.c#L123) for this, it uses three tabs relative to its parent line.  I'm honestly not sure if two or three extra tabs is the desired style for this.